Task7_2/NorthwindHandler/Services/ResponseGenerator.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using NorthwindHandler.Models;
+using NorthwindHandler.Services.Abstractions;
+
+namespace NorthwindHandler.Services
+{
+    public class ResponseGenerator : IResponseGenerator
+    {
+        private readonly IEnumerable<IReportGenerator> _reportGenerators;
+        private const ReportFormat DefaultFormat = ReportFormat.Excel;
+
+        public ResponseGenerator(IEnumerable<IReportGenerator> reportGenerators)
+        {
+            _reportGenerators = reportGenerators;
+        }
+
+        public void GenerateResponse(HttpContext context, IEnumerable<OrderModel> orders)
+        {
+            var acceptHeaderValue = context.Request.Headers["Accept"];
```

Comment:
It is not good place for http dependencies.
You could create resolver for report format and retrieve it in handler